lms/validation/_base.py

```diff
@@ -94,6 +95,18 @@ def parse(self, *args, **kwargs):
     def _handle_error(error, _req, _schema, *, error_status_code, error_headers):
         raise ValidationError(messages=error.messages) from error
 
+    @marshmallow.pre_load
+    def _decode_jwt(self, data, **_kwargs):
```

Comment:
This makes all schemas used to get "parsed_params" LTI1.3 compatible using the params from the id_token.

Reply:
This kind of really seems like it's in the wrong place to me, or I don't understand, or both? This seems like it would apply way too broadly?

I'm also very confused about the relationship between the code in `security` adding this stuff, the eventual schemas that use it etc...

I suspect honestly I'm really missing the point here. Might be time to call the review and have a chat.

Reply:
We had a chat and I think the conclusions are:

* Although all `request.parsed_params` are `PyramidRequestSchema`
* ...and all `LTILaunchBlahSchema` are `PyramidRequestSchema`
* ... not all `PyramidRequestSchema` have anything to do with `LTILaunch...` - plenty are used for internal stuff, or Canvas etc. So this is more specific.
* `lms.validation._lti.LTIAuthParamsSchema` looks like a good home, but it's not a child of `PyramidRequestSchema`
* ... but every child of it is and it's not used independently

So:

 * `LTIAuthParamsSchema` should probably be a child of `PyramidRequestSchema`
 * All its children can then stop having multiple inheritance
 * And this method can move there